lib/puppet/provider/package/pip.rb

```diff
@@ -131,4 +146,33 @@ def lazy_pip(*args)
   def install_options
     join_options(@resource[:install_options])
   end
+
+  def latest_with_new_pip
+    # Less resource intensive approach for pip version 1.5.4 and above
+    execpipe ["#{self.class.pip_cmd}", "install", "#{@resource[:name]}==versionplease"] do |process|
```

Comment:
Ok, looking at how you're using it, seems like it's probably easier to parse install.